Week4/homework/ex1-aggregation/ex1-aggregation.js

```diff
@@ -0,0 +1,74 @@
+require('dotenv').config();
+
+const { MongoClient } = require('mongodb');
+
+const client = new MongoClient(process.env.MONGODB_URL);
+
+async function getTotalPopulationForGivenCountryPerYear(client, country) {
+  const pipeline = [
+    {
+      $match: {
+        Country: country,
+      },
+    },
+    {
+      $group: {
+        _id: '$Year',
+        countPopulation: {
+          $sum: {
+            $add: ['$M', '$F'],
+          },
+        },
+      },
+    },
+    {
+      $sort: {
+        _id: 1,
+      },
+    },
+  ];
+
+  const aggCursor = await client
+    .db('databaseWeek4')
+    .collection('population_pyramid')
+    .aggregate(pipeline);
+
+  await aggCursor.forEach(console.log);
+}
+
+getTotalPopulationForGivenCountryPerYear(client, 'Netherlands');
+
+async function getContinentInformation(client, year, age) {
+  const pipeline = [
+    {
+      $match: {
+        $or: [
+          { Country: { $regex: '^[A-Z]+$' } }, // Matches all-capital letters but if there is a space, it will be a problem.
```

Comment:
Might help you out here.. 😄 the `$` in regex designates the end of a string. So now it will match anything starting with 1 or more characters that are in the set of characters `A-Z`.

```suggestion
          { Country: { $regex: '^[A-Z]+' } }, // Matches all-capital letters but if there is a space, it will be a problem.
```